File: 3600-find-the-k-th-character-in-string-game-i/find-the-k-th-character-in-string-game-i.py
class Solution:
    def kthCharacter(self, k: int) -> str:
        word="a"
        while len(word)<k:
            temp=""
            for i in word:
                new=(ord(i)+1)%(ord("z")+1)
                temp+=chr(new)
            word=word+temp
            print(word)
        return word[k-1]
        

File: 3600-find-the-k-th-character-in-string-game-i/find-the-k-th-character-in-string-game-i.java
import java.util.*;
class Solution {
    public char kthCharacter(int k) {
        List<Character> list=new ArrayList<>();
        list.add('a');
        while(list.size()<k){
            int sz=list.size();
            for(int i=0;i<sz;i++){
                char c=list.get(i);
                char newChar = (char)(((c - 'a' + 1) % 26) + 'a');

                list.add(newChar);
            }

        }
        System.out.println(list);
        char answer=list.get(k-1);

        return answer;
    }
}